egs/wsj/s5/steps/nnet3/compute_and_adjust_priors.py

```diff
@@ -0,0 +1,255 @@
+
+    num_tasks_to_process = 1
+    # If True, it is assumed different examples used to train multiple
+    # tasks in the output layer.
+    use_multitask_egs = False
+    if (os.path.exists('{0}/info/num_tasks'.format(egs_dir))):
+        num_tasks_to_process = int(open('{0}/info/num_tasks'.format(
+                                        egs_dir)).readline())
+    if num_tasks_to_process > 1:
+        use_multitask_egs = True
+
+    avg_post_vec_file = train_lib.common.compute_average_posterior(
+        dir=args.dir, iter=args.output_name, egs_dir=egs_dir, model=args.model,
+        num_archives=num_archives,
+        left_context=left_context, right_context=right_context,
+        prior_subset_size=args.prior_subset_size, run_opts=run_opts,
+        output_name=args.output_name,
+        use_multitask_egs=use_multitask_egs)
+
+    if args.readjust_priors:
+        logger.info("Re-adjusting priors based on computed posteriors")
+        if args.model is not None:
+            if args.ali_dir is None:
+                init_model = "{dir}/{init_mdl}.raw".format(dir=args.dir, init_mdl=args.model)
+            else:
+                train_lib.acoustic_model.prepare_acoustic_model(dir=args.dir,
+                                                                alidir=args.ali_dir,
+                                                                run_opts=run_opts,
+                                                                model=args.model)
+                init_model = "{dir}/{model}.mdl".format(dir=args.dir, model=args.model)
+
+        else:
+          if args.ali_dir is None:
+              # If ali_dir is not specified, it assumes the model contains transition model.
+              init_model = "{dir}/combined.mdl".format(dir=args.dir)
+          else:
+              train_lib.acoustic_model.prepare_acoustic_model(dir=args.dir,
+                                                              alidir=args.ali_dir,
+                                                              run_opts=run_opts,
+                                                              model="final")
+              init_model = "{dir}/final.mdl".format(dir=args.dir)
+
+        final_model = "{dir}/{target_mdl}.mdl".format(dir=args.dir, target_mdl=args.readjusted_model)
+        train_lib.common.adjust_am_priors(args.dir, init_model,
+                                          avg_post_vec_file, final_model,
+                                          run_opts)
+def main():
+    [args, run_opts] = get_args()
+    try:
+        background_process_handler = common_lib.BackgroundProcessHandler(
+            polling_time=args.background_polling_time)
+        compute_and_adjust_priors(args, run_opts, background_process_handler)
+        background_process_handler.ensure_processes_are_done()
+    except Exception as e:
+        if args.email is not None:
+            message = ("Training session for experiment {dir} "
```

Comment:
Add a more relevant message. Or you can remove email entirely. Its only used in the training scripts.